Code/src/main/java/nl/utwente/viskell/ui/components/Connection.java

```diff
@@ -83,38 +87,26 @@ public void setErrorState(boolean error) {
     }
     
     /**
-     * Get the optional output anchor on the other side of
-     * the provided input anchor in this Connection.
-     * 
-     * @param anchor on this side of the connection  
-     * @return the output anchor if it exists in this connection.
+     * @return the output anchor of this connection.
      */
-    public Optional<OutputAnchor> getOppositeAnchorOf(InputAnchor anchor) {
-        return this.endAnchor.filter(ia -> ia == anchor).flatMap(x -> this.startAnchor);
+    public OutputAnchor getStartAnchor() {
+        return this.startAnchor;
     }
 
     /**
-     * Get the optional output anchor on the other side of
-     * the provided input anchor in this Connection.
-     * 
-     * @param anchor on this side of the connection  
-     * @return the input anchor if it exists in this connection.
+     * @return the input anchor of this connection.
      */
-    public Optional<InputAnchor> getOppositeAnchorOf(OutputAnchor anchor) {
-        return this.startAnchor.filter(oa -> oa == anchor).flatMap(x -> this.endAnchor);
+    public InputAnchor getEndAnchor() {
+        return this.endAnchor;
     }
     
     /**
      * Handles the upward connections changes through an connection.
      * Also perform typechecking for this connection.
-     * @param input the input anchor of this change propagation.
      */
-    public void handleConnectionChangesFrom(InputAnchor input) {
-        if (!this.startAnchor.isPresent()) {
-            return;
-        }
-        
-        OutputAnchor output = this.startAnchor.get();
+    public void handleConnectionChangesUpwards() {
+        InputAnchor input = this.endAnchor;
+        OutputAnchor output = this.startAnchor;
```

Comment:
Is this basically a rename of the `endAnchor` and `startAnchor` variables or is there something else going on?